naga/src/front/spv/image.rs

```diff
@@ -526,7 +526,20 @@ impl<I: Iterator<Item = u32>> super::Frontend<I> {
             image_ops ^= bit;
         }
 
-        let si_lexp = self.lookup_sampled_image.lookup(sampled_image_id)?;
+        let si_lexp = self.lookup_sampled_image.lookup(sampled_image_id);
+        let si_lexp = match si_lexp {
```

Comment:
I think you could just say

```
let si_lexp = match self.lookup_sampled_image.get(sampled_image_id) {
    Some(lookup) => ...
    None => ...
}
```

here and skip the whole `InvalidId` error because it's irrelevant